internal/service/handlers/verify_passport.go

```diff
@@ -169,13 +191,64 @@ func fulfillPassportScanEvent(r *http.Request, balance data.Balance) error {
 		return errors.New("inconsistent state: balance has no country, event type is active, but no open event was found")
 	}
 
-	_, err = EventsQ(r).
+	event, err = EventsQ(r).
 		FilterByID(event.ID).
 		Update(data.EventFulfilled, nil, nil)
 
+	if !evTypePassport.AutoClaim || !country.ReserveAllowed || country.Reserved >= country.ReserveLimit {
+		return err
```

Comment:
nil error, the error above is not checked; why would we have this condition here?